doc/relevancy.md

```diff
@@ -0,0 +1,460 @@
+> nice and stable measure.
+
+If we want to boost the popularity of a query then we do so by 0.001
+(`POPULARITY_OFFSET = 0.001` in [popularity.rb][]).  It is not clear
+where this number came from.
+
+We also boost using `boost_mode: :multiply` (multiply the `_score`
+with the function result (default)).
+
+We store page traffic data in the `page-traffic` index.  For each page
+we store its rank relative to other pages on GOV.UK in the `rank_14`
+field and its number of page views in the `vc_14` field.
+
+#### Recency
+
+This is an implementation of [this curve][], and is applied to
+documents of the "announcement" type in the [booster.rb][] file.  It
+serves to increase the score of new documents and decrease the score
+of old documents.
+
+Only documents of `search_format_types` 'announcement' are affected by
+recency boosting.
+
+The curve came from the solr documentation, but the link to the
+original source is broken, so it is unclear why this particular curve
+was chosen.
+
+#### Properties
+
+This is defined by the [boosting.yml][] file, and is applied in the
+[booster.rb][] file.
+
+For example, if a document has `is_historic: true` we downrank it, so
+it's less likely to appear at the top of search results.
+
+It is unclear where the numbers came from.
+
+### Best and worst bets
+
+Some search queries have perfect answers.  For example, a search for
+"HM Treasury" should have the HM Treasury organisation page as the top
+result.
+
+To handle these cases we have a concept called "best bets".
+
+A "best bet" for a search query is a result that should be shoehorned
+into the top of the results list for that query.  It's a way of
+boosting that is a surefire route to having your document top the
+results for a given query.
+
+Best bets are managed with the [Search Admin][] application.
+
+Example best bets:
+
+```
+Algeria: /foreign-travel-advice/algeria
+Citizenship: /becoming-a-british-citizen
+```
+
+There's also a concept called "worst bets".  These are useful if our
+search results are so bad that we have to manually downrank a result
+for a given query.  For example, "vehicle" shouldn't result in
+"unicycle" being the top result.
+
+Best and worst bets are implemented in the [best_bets.rb][] file.
+
+### Stopwords
+
+Stopwords are words like "is", "and", "an", and "a".  They're words we
+filter out when processing a search query, since they're so common
+they're not useful.  They also enable Elasticsearch to maintain a
+smaller index.
+
+The stopwords themselves are provided by Elasticsearch.  There's
+nothing in the Search API repo that looks at a list of omitted terms.
+This is handled by the default [stop token filter][].
+
+**History of stopwords**
+
+There has been some talk of not including stopwords in the past
+([0fe6e52][], May 2015) during this time a `no_stop` method was
+implemented that seems to prevent stopwords from being used.  Again this
+seems to be served by Elasticsearch because it's not defined in Search
+API.
+
```

Comment:
The paragraph below seems to belong elsewhere as it's written in the first person and is a discussion on what we might do in the future. Maybe changing the title from `History of Stopwords` to something else would help. Eg `Future improvements around stopwords` sort of thing. Might be better to remove the `I'm not sure ` and say `It's not immediately clear how we'd go about changing this, but it's likely that we'd need to look into what words could be useful to omit/not omit`

Reply:
Thanks, I'll add a header to this section **Potential improvements to stopwords**.